json-tool/public/assets/js/features/codegen/codegen.js

```
@@ -0,0 +1,267 @@
+  saveGlobalState();
+}
+
+function copyCodeOutput(tabId) {
+  const codePre = document.querySelector(`#${tabId} .code-output`);
+  copyToClipboard(codePre.textContent, "Code copied");
+}
+
+function generateTypeScript(obj, interfaceName) {
+  let result = `interface ${interfaceName} {\n`;
+  for (let key in obj) {
+    if (!obj.hasOwnProperty(key)) continue;
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)